src/lexitools/commands/correspondences.py

```diff
@@ -31,150 +33,195 @@ def register(parser):
     args.log.info('Loaded the wordlist')
 
+    concepts_kept = set((a,b) for idx, a,b in lex.iter_rows('concept', 'concepticon'))
+    args.log.info("A total of {} concepts were used".format(len(concepts_kept)))
+
+
+    # Record phoneme frequency
     G = nx.Graph()
     for idx, tokens, language in lex.iter_rows('tokens', 'doculect'):
-        for sound in [t.split('/')[1] if '/' in t else t for t in tokens]:
-            try:
-                #G.node[sound]['frequency'] += 1
-                G.node[sound]['language'].add(language)
-            except:
-                G.add_node(
-                        sound, 
-                        frequency=1, 
-                        language=set([language]),
-                        name=bipa[sound].name, 
-                        color=color[sound],
-                        asjp=asjp[sound],
-                        sca=sca[sound],
-                        type=bipa[sound].type
-                        )
-                
+        for sound in iter_phonemes(tokens):
+            if sound not in "+*-":
+                sound = sound_class[sound]
+                try:
+                    G.node[sound]['frequency'][language] += 1
+                except:
+                    G.add_node(sound, frequency=Counter({language: 1}))
+
     # add the dummy node for gaps, in case we want to use it
-    G.add_node('-',frequency=0, language=set(lex.cols))
+    # G.add_node('-', frequency=Counter())
 
-    for node, data in G.nodes(data=True):
-        data['language'] = ', '.join(sorted(data['language']))
+    # for node, data in G.nodes(data=True):
+    #     data['language'] = ', '.join(sorted(data['language']))
 
+    n_lang = len(lex.cols)
+    tot_pairs = (n_lang * (n_lang - 1)) / 2
 
-    for lA, lB in progressbar(combinations(lex.cols, r=2)):
+    # Record correspondences
+    for lA, lB in progressbar(combinations(lex.cols, r=2), total=tot_pairs):
         for idxA, idxB in lex.pairs[lA, lB]:
-            tokensA, tokensB = (
-                    [t.split('/')[1] if '/' in t else t for t in lex[idxA, 'tokens']],
-                    [t.split('/')[1] if '/' in t else t for t in lex[idxB, 'tokens']]
-                    )
+            tokensA = list(iter_phonemes(lex[idxA, 'tokens']))
+            tokensB = list(iter_phonemes(lex[idxB, 'tokens']))
             langA, langB = lex[idxA, 'doculect'], lex[idxB, 'doculect']
-            classesA, classesB = lex[idxA, 'classes'], lex[idxB, 'classes']
-
-            # check for edit dist == 1
-            if edit_dist(
-                    classesA, 
-                    classesB
-                        ) <= args.threshold:
-                pair = Pairwise(tokensA, tokensB)
+            # classesA, classesB = lex[idxA, 'classes'], lex[idxB, 'classes']
+
+            # check for edit dist == 1giot
+            token_strs = ' '.join(tokensA), ' '.join(tokensB)
+            potential_corresp = lingpy.edit_dist(*token_strs) <= args.threshold
```

Comment:
Problem with this solution is: you convert to ASJP etc. only later below. But edit distance for ASJP strings will be more often == 1 if you convert to ASJP strings before. For this reason, the conversion if asjp is chosen as sound class, should be chosen when extracting the tokens. So the sound class conversion should take place now.

Reply:
I agree, changing it right now :)

Reply:
Ok, so I can convert to asjp (or other system) earlier, but only for the edit distances. Pairwise complains if I run it on ASJP strings, for a pair like 'w o 6' (originally 'u/w o ²¹⁴') 'N E 5' (originally 'ŋ æ̃ ⁵⁵'), I get:

~~~
ValueError: [!] your sequence contains only unknown characters
~~~